keras_cv/layers/preprocessing/random_rotation.py

```diff
@@ -0,0 +1,224 @@
+        bbox_dtype = bounding_boxes.dtype
+        if self.bounding_box_format is None:
+            raise ValueError(
+                "bounding_box_format has not been specified during layer "
+                "initialization. Please refer docstring for more information."
+            )
+        else:
+            bounding_boxes = bounding_box.convert_format(
+                bounding_boxes, source=self.bounding_box_format, target="xyxy"
+            )
+        image = tf.expand_dims(image, 0)
+        image_shape = tf.shape(image)
+        h = image_shape[H_AXIS]
+        w = image_shape[W_AXIS]
+        # origin coordinates, all the points on the image are rotated around
+        # this point
+        origin_x, origin_y = int(h / 2), int(w / 2)
+        angle = transformation["angle"]
+        angle = -angle
+        # calculate coordinates of all four corners of the bounding box
+        point = tf.stack(
+            [
+                tf.stack([bounding_boxes[:, 0], bounding_boxes[:, 1]], axis=1),
+                tf.stack([bounding_boxes[:, 2], bounding_boxes[:, 1]], axis=1),
+                tf.stack([bounding_boxes[:, 2], bounding_boxes[:, 3]], axis=1),
+                tf.stack([bounding_boxes[:, 0], bounding_boxes[:, 3]], axis=1),
+            ],
+            axis=1,
+        )
+        # point_x : x coordinates of all corners of the bounding box
+        point_x = tf.gather(point, [0], axis=2)
+        # point_y : y cordinates of all corners of the bounding box
+        point_y = tf.gather(point, [1], axis=2)
+        # rotated bbox coordinates
+        # new_x : new position of x coordinates of corners of bounding box
+        new_x = (
+            origin_x
+            + tf.multiply(
+                tf.cos(angle), tf.cast((point_x - origin_x), dtype=tf.float32)
+            )
+            - tf.multiply(
+                tf.sin(angle), tf.cast((point_y - origin_y), dtype=tf.float32)
+            )
+        )
+        # new_y : new position of y coordinates of corners of bounding box
+        new_y = (
+            origin_y
+            + tf.multiply(
+                tf.sin(angle), tf.cast((point_x - origin_x), dtype=tf.float32)
+            )
+            + tf.multiply(
+                tf.cos(angle), tf.cast((point_y - origin_y), dtype=tf.float32)
+            )
+        )
+        # rotated bbox coordinates
+        out = tf.concat([new_x, new_y], axis=2)
+        # find readjusted coordinates of bounding box to represent it in corners
+        # format
+        min_cordinates = tf.math.reduce_min(out, axis=1)
+        max_cordinates = tf.math.reduce_max(out, axis=1)
+        bboxes_out = tf.concat([min_cordinates, max_cordinates], axis=1)
+        # cordinates cannot be float values, it is casted to int32
+        bboxes_out = bounding_box.convert_format(
+            bboxes_out, source="xyxy", target=self.bounding_box_format
+        )
+        bboxes_out = tf.cast(bboxes_out, bbox_dtype)
```

Comment:
I agree these casts make the behavior less surprising, but I wonder if they are worth it.  Perhaps casting to `self.compute_dtype` is more consistent.

Reply:
casting to `self.compute_dtype` converts it bbox output to float, even if bbox input is int. And for bbox coordinates -float is not expected.

Reply:
Actually I'm not sure that I agree with this: float is a reasonable datatype for bounding boxes.  rel_* bounding boxes are floats, many COCO annotations have coordinates set as floats, rescaling operations output bounding boxes as floats. 

My concern is that if we don't cast them in all layers the behavior is now inconsistent

Reply:
Actually I'm not sure that I agree with this: float is a reasonable datatype for bounding boxes.  rel_* bounding boxes are floats, many COCO annotations have coordinates set as floats, rescaling operations output bounding boxes as floats. 

My concern is that if we don't cast them in all layers the behavior is now inconsistent

Reply:
Actually I'm not sure that I agree with this: float is a reasonable datatype for bounding boxes.  rel_* bounding boxes are floats, many COCO annotations have coordinates set as floats, rescaling operations output bounding boxes as floats. 

My concern is that if we don't cast them in all layers the behavior is now inconsistent

Reply:
If the input format is float, we will be returning float, but if the input format is int, then returning a float value would not make sense to me. If the bboxes are int values for an image of size 512x512, what does a float value of coordinates indicate? what location on the image?